File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ChargeItemDefinition.py
from typing import Optional, List
from pydantic import *
from ..base import *

class ChargeItemDefinition_PropertyGroup_PriceComponent(BackboneElement):
    code: Optional[CodeableConcept] = None
    type: str
    amount: Optional[Money] = None
    factor: Optional[float] = None

class ChargeItemDefinition_PropertyGroup(BackboneElement):
    applicability: Optional[List[Reference]] = None
    price_component: Optional[List[ChargeItemDefinition_PropertyGroup_PriceComponent]] = None

class ChargeItemDefinition_Applicability(BackboneElement):
    language: Optional[str] = None
    expression: Optional[str] = None
    description: Optional[str] = None

class ChargeItemDefinition(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    approval_date: Optional[str] = None
    property_group: Optional[List[ChargeItemDefinition_PropertyGroup]] = None
    instance: Optional[List[Reference]] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    title: Optional[str] = None
    derived_from_uri: Optional[List[str]] = None
    status: str
    url: str
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = None
    last_review_date: Optional[str] = None
    replaces: Optional[List[str]] = None
    part_of: Optional[List[str]] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = None
    applicability: Optional[List[ChargeItemDefinition_Applicability]] = None
    effective_period: Optional[Period] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/ChargeItemDefinition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class ChargeItemDefinition : DomainResource, IResource
{
    public string? Description { get; set; }
    public string? Date { get; set; }
    public string? Publisher { get; set; }
    public string? ApprovalDate { get; set; }
    public ChargeItemDefinition_PropertyGroup[]? PropertyGroup { get; set; }
    public Base.ResourceReference[]? Instance { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public Base.UsageContext[]? UseContext { get; set; }
    public string? Copyright { get; set; }
    public bool? Experimental { get; set; }
    public string? Title { get; set; }
    public string[]? DerivedFromUri { get; set; }
    public required string Status { get; set; }
    public required string Url { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? LastReviewDate { get; set; }
    public string[]? Replaces { get; set; }
    public string[]? PartOf { get; set; }
    public string? Version { get; set; }
    public Base.ContactDetail[]? Contact { get; set; }
    public ChargeItemDefinition_Applicability[]? Applicability { get; set; }
    public Base.Period? EffectivePeriod { get; set; }

    public class ChargeItemDefinition_PropertyGroup_Applicability : BackboneElement
    {
        public string? Language { get; set; }
        public string? Expression { get; set; }
        public string? Description { get; set; }
    }

    public class ChargeItemDefinition_PropertyGroup_PriceComponent : BackboneElement
    {
        public Base.CodeableConcept? Code { get; set; }
        public required string Type { get; set; }
        public Base.Money? Amount { get; set; }
        public string? Factor { get; set; }
    }

    public class ChargeItemDefinition_PropertyGroup : BackboneElement
    {
        public ChargeItemDefinition_PropertyGroup_Applicability[]? Applicability { get; set; }
        public ChargeItemDefinition_PropertyGroup_PriceComponent[]? PriceComponent { get; set; }
    }

    public class ChargeItemDefinition_Applicability : BackboneElement
    {
        public string? Language { get; set; }
        public string? Expression { get; set; }
        public string? Description { get; set; }
    }
}